src/carriers/zfp_portmonitor/CMakeLists.txt

```diff
@@ -8,24 +8,34 @@ yarp_prepare_plugin(zfp TYPE ZfpMonitorObject
                         DEPENDS "CREATE_OPTIONAL_CARRIERS;ENABLE_yarpcar_portmonitor")
 
 if(NOT SKIP_zfp)
-  set(CMAKE_INCLUDE_CURRENT_DIR ON)
-  get_property(YARP_OS_INCLUDE_DIRS TARGET YARP_OS PROPERTY INCLUDE_DIRS)
-  get_property(YARP_sig_INCLUDE_DIRS TARGET YARP_sig PROPERTY INCLUDE_DIRS)
-  include_directories(${YARP_OS_INCLUDE_DIRS} ${YARP_sig_INCLUDE_DIRS})
-  include_directories(SYSTEM ${ZFP_INCLUDE_DIRS})
-
-  yarp_add_plugin(yarp_pm_zfp zfpPortmonitor.cpp
-                              zfpPortmonitor.h)
-  target_link_libraries(yarp_pm_zfp YARP_OS YARP_sig ${ZFP_LIBRARIES})
-
-  yarp_install(TARGETS yarp_pm_zfp
-               EXPORT YARP
-               COMPONENT runtime
-               LIBRARY DESTINATION ${YARP_DYNAMIC_PLUGINS_INSTALL_DIR}
-               ARCHIVE DESTINATION ${YARP_STATIC_PLUGINS_INSTALL_DIR})
-  yarp_install(FILES zfp.ini
-               COMPONENT runtime
-               DESTINATION ${YARP_PLUGIN_MANIFESTS_INSTALL_DIR})
-
-  set_property(TARGET yarp_pm_zfp PROPERTY FOLDER "Plugins/Port Monitor")
+
+    if(ZFP_FOUND)
```

Comment:
I like the idea of grouping devices together and showing/hiding them as a whole.
I changed it because right now in `devel`, when the library is not  found the output says:
```
 CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
 Please set them or make sure they are set and tested correctly in the CMake files:
 ZFP_INCLUDE_DIR (ADVANCED)
    used as include directory in directory /usr/local/src/robot/ROBOT_CODE/01_yarp/yarp/src/carriers/zfp_portmonitor
    used as include directory in directory /usr/local/src/robot/ROBOT_CODE/01_yarp/yarp/src/carriers/zfp_portmonitor
    used as include directory in directory /usr/local/src/robot/ROBOT_CODE/01_yarp/yarp/src/carriers/zfp_portmonitor
    used as include directory in directory /usr/local/src/robot/ROBOT_CODE/01_yarp/yarp/src/carriers/zfp_portmonitor
    used as include directory in directory /usr/local/src/robot/ROBOT_CODE/01_yarp/yarp/src/carriers/zfp_portmonitor
    used as include directory in directory /usr/local/src/robot/ROBOT_CODE/01_yarp/yarp/src/carriers/zfp_portmonitor
    used as include directory in directory /usr/local/src/robot/ROBOT_CODE/01_yarp/yarp/src/carriers/zfp_portmonitor
    used as include directory in directory /usr/local/src/robot/ROBOT_CODE/01_yarp/yarp/src/carriers/zfp_portmonitor
    used as include directory in directory /usr/local/src/robot/ROBOT_CODE/01_yarp/yarp/src/carriers/zfp_portmonitor
    used as include directory in directory /usr/local/src/robot/ROBOT_CODE/01_yarp/yarp/src/carriers/zfp_portmonitor
```
Which is confusing and not meaningful.

Anyway, it is not true that using `zfp` as a dependency
>   the output tells you all you need to know...

What actually happens is that the plugin will be hidden and, if you really look carefully the cmake will only write
`--  --- plugin yarpcar_zfp: disabled (dependencies unsatisfied)`
without telling what is missing and what has to be done to satisfy this dependency.

With the patch proposed the error message will be:
```
ZFP carrier: ZFP library not found. 
 Please download the 'ZFP' library from 'http://computation.llnl.gov/projects/floating-point-compression',
 build it using cmake and install it. Set the environment variable 'ZFP_DIR' to the installation folder. 
```
Which is more clear and useful.

I like the idea of hiding some flags depending of options to shorten the cmake, like using `CREATE_OPTIONAL_CARRIERS` as a selector, but I do believe that 'logical selection' shall hide things while missing dependencies shall not, they have to raise clear error to the user instead hiding stuff under the carpet.

All the devices I know uses works the same as this plugin, and historically this is always been the way they worked and how the user expect them to be.
All the devices always appears in the list (once the `complie_XXX` is enabled) and when the single device has been selected, dependency errors are thrown.